docs/Additional-Activation-OTP.md

```diff
@@ -0,0 +1,168 @@
+   
+   1. The user enters login and password
+   1. Mobile application then initializes the custom activation with provided login and password.
+
+1. Intermediate Server Application receives a custom activation request, with login and password:
+
+   1. The login and password is verified by the Intermediate Server Application
+   1. If everything's right, then Intermediate Server Application creates activation by calling [`createACtivation`](https://github.com/wultra/powerauth-server/blob/develop/docs/SOAP-Service-Methods.md#method-createactivation). The activation must be set to the OTP_USED state.
+   1. Intermediate Server Application generates random activation OTP and update the activation record, by calling [`updateActivationOtp`](https://github.com/wultra/powerauth-server/blob/develop/docs/SOAP-Service-Methods.md#method-updateactivationotp) method.
+   1. At the same time, Intermediate Server Application initiates the delivery of activation OTP.
+   
+1. Back in the mobile application:
+
+   1. The mobile application receives the response from the server and completes the keys-exchange on the mobile side.
+   1. The user waits for OTP delivery via the out-of-band channel.
+   1. The user retypes OTP.
+   1. Mobile application then commits the activation with OTP, by calling a custom RESTful endpoint, protected with our [ECIES encryption](End-To-End-Encryption.md) scheme.
+
+1. Intermediate Server Application then receives the commit request
+
+   1. Decrypts OTP from the request
+   1. Commits the activation by calling PowerAuth Server's [`commitActivation`](https://github.com/wultra/powerauth-server/blob/develop/docs/SOAP-Service-Methods.md#method-commitactivation) method.
+
+After the response from the commit is received on the mobile side, the application can check the  whether the activation's state is ACTIVE.
+
+### Confirm activation recovery with OTP
+
+The confirmation of activation recovery is very similar to custom activation confirmation.
+
+1. In the mobile application:
+   
+   1. The user enters recopvery code and PUK
+   1. Mobile application then initializes the recovery activation with provided code and PUK.
+
+1. Intermediate Server Application receives a recovery activation request:
+
+   1. The login and password is verified by the PowerAuth Server, by calling [`recoveryCodeActivation`](https://github.com/wultra/powerauth-server/blob/develop/docs/SOAP-Service-Methods.md#method-recoverycodeactivation).
+   1. Intermediate Server Application generates random activation OTP and update the activation record, by calling [`updateActivationOtp`](https://github.com/wultra/powerauth-server/blob/develop/docs/SOAP-Service-Methods.md#method-updateactivationotp) method.
+   1. At the same time, Intermediate Server Application initiates the delivery of activation OTP.
+   
+1. Back in the mobile application:
+
+   1. The mobile application receives the response from the server and completes the keys-exchange on the mobile side.
```

Comment:
The `keys-exchange` should probably be rather `key exchange`, this is a term in cryptography.